exporter.go

```diff
@@ -24,43 +26,98 @@ func getEnv(key, defaultValue string) string {
 	return value
 }
 
+func getEnvInt(key string, defaultValue int) int {
+	value, ok := os.LookupEnv(key)
+	if !ok {
+		return defaultValue
+	}
+	i, err := strconv.ParseInt(value, 10, 64)
+	if err != nil {
+		log.Fatalf("Environment variable value for %s must be an int: %v", key, err)
+	}
+	return int(i)
+}
+
 func getEnvBool(key string, defaultValue bool) bool {
 	value, ok := os.LookupEnv(key)
 	if !ok {
 		return defaultValue
 	}
 	b, err := strconv.ParseBool(value)
 	if err != nil {
-		log.Fatalf("Environment Variable value for %s must be a boolean", key)
+		log.Fatalf("Environment variable value for %s must be a boolean: %v", key, err)
 	}
 	return b
 }
 
+func getEnvDuration(key string, defaultValue time.Duration) time.Duration {
+	value, ok := os.LookupEnv(key)
+	if !ok {
+		return defaultValue
+	}
+	d, err := time.ParseDuration(value)
+	if err != nil {
+		log.Fatalf("Environment variable value for %s must be a duration: %v", key, err)
+	}
+	return d
+}
+
+func createClientWithRetries(getClient func() (interface{}, error), retries int, retryInterval time.Duration) (interface{}, error) {
+	var err error
+	var nginxClient interface{}
+	for retryNum := retries; retryNum >= 0; retryNum-- {
```

Comment:
for loops, it is better to use simple variables like `i`

Reply:
also, consider using the following body:
```golang
		nginxClient, err = getClient()
		if err == nil {
			return nginxClient, nil
		}

		if i > 0 {
			log.Printf("Could not create Nginx Client: %v. Retrying in %v...", err, retryInterval)
			time.Sleep(retryInterval)
		}
```

Note that if instead of using else, it is always better to return if possible